x/ecocredit/server/helpers.go

```diff
@@ -5,47 +5,46 @@ import (
 	"encoding/binary"
 	"fmt"
 
-	"github.com/cockroachdb/apd/v2"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 	"github.com/regen-network/regen-ledger/types/math"
 )
 
-func getDecimal(store sdk.KVStore, key []byte) (*apd.Decimal, error) {
+func getDecimal(store sdk.KVStore, key []byte) (*math.Dec, error) {
 	bz := store.Get(key)
 	if bz == nil {
-		return apd.New(0, 0), nil
+		tmp := math.NewDecFromInt64(0)
+		return &tmp, nil
 	}
 
-	value, _, err := apd.NewFromString(string(bz))
+	value, err := math.NewDecFromString(string(bz))
 	if err != nil {
 		return nil, sdkerrors.Wrap(err, fmt.Sprintf("can't unmarshal %s as decimal", bz))
 	}
 
-	return value, nil
+	return &value, nil
 }
 
-func setDecimal(store sdk.KVStore, key []byte, value *apd.Decimal) {
+func setDecimal(store sdk.KVStore, key []byte, value *math.Dec) {
 	// always remove all trailing zeros for canonical representation
-	value, _ = value.Reduce(value)
+	*value, _ = value.Reduce()
 
 	if value.IsZero() {
 		store.Delete(key)
 	} else {
 		// use floating notation here always for canonical representation
-		str := value.Text('f')
-		store.Set(key, []byte(str))
+		store.Set(key, []byte(value.String()))
 	}
 }
 
-func getAddAndSetDecimal(store sdk.KVStore, key []byte, x *apd.Decimal) error {
+func getAddAndSetDecimal(store sdk.KVStore, key []byte, x *math.Dec) error {
 	value, err := getDecimal(store, key)
 	if err != nil {
 		return err
 	}
 
-	err = math.Add(value, value, x)
+	*value, err = value.Add(*x)
```

Comment:
We generally don't want to use pointers with these decimal struct. Instead let's add a return value.

Reply:
in that case - should i just change all the helper.go functions to deal with values rather than pointers? or do we want these helper functions to receive/return pointers?

Reply:
All functions dealing with the new `Dec` struct should not use pointers I believe. That's at least the intention.